lib/app/item.ex

```diff
@@ -2,23 +2,26 @@ defmodule App.Item do
   use Ecto.Schema
   import Ecto.Changeset
   import Ecto.Query
-  alias App.{Repo, Tag, ItemTag, Person}
+  alias App.{Repo, Tag, ListItem, ItemTag, Person}
+  alias App.List, as: L
   alias __MODULE__
 
   schema "items" do
     field :status, :integer
     field :text, :string
+    field :item_lists, {:array, :string}, virtual: true
 
     belongs_to :people, Person, references: :person_id, foreign_key: :person_id
     many_to_many(:tags, Tag, join_through: ItemTag, on_replace: :delete)
+    many_to_many(:lists, L, join_through: ListItem, on_replace: :delete)
```

Comment:
I honestly don't feel the need to add these definitions to the schema.
I _understand_ that it's the "Ecto way" of doing things.
But it doesn't add anything to to underlying Postgres table or improve query performance.
So treating it as "documentation" for now. 👍

Reply:
These `many_to_many` definitions allow Phoenix/Ecto to know what to do when an item is removed from a list. The `on_replace: :delete` value will remove the association in the join table automatically when an item is removed

Reply:
Yeah, got it. But from a UI/UX perspective `LiveView` is the the source of truth synching the data between clients.
And from `Elixir/Phoenix` perspective the DB (Postgres) is the single source of truth. 
These `Ecto` annotations are just forcing us to use the ORM, they don't speed up the queries extracting the data.
I would _much_ rather write portable `SQL` code that I can easily test in my `pgAdmin` when I need to analyse query performance than use a [DSL](https://en.wikipedia.org/wiki/Domain-specific_language) that hides the "truth" from me and is a pain to debug.

Muuuuch prefer the approach that IHP has taken: https://ihp.digitallyinduced.com/Features
![image](https://user-images.githubusercontent.com/194400/194917989-5eed5caf-83a4-4d39-9daa-6fa21147a08a.png)

If I could write **_zero_ `Ecto` queries** I'd be **_so_ much happier**.
But I see the benefits accrued in [**`Fields`**](https://github.com/dwyl/fields) using `Ecto Types`. 💭 

Like I said above, If I could avoid having to _manually_ define these `many_to_many` relationships in an ORM and just have them in the SQL and let `Ecto` infer them from the SQL Schema (not `Elixir` code) at compile time it would make me _so_ much happier. That way the Database Schema (`SQL`) is the single source of truth, not the ORM. 

Anyway. Perhaps not the place for this kind of "rant". 🙄